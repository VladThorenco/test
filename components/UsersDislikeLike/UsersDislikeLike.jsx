import './UsersDislikeLike.scss';
import React, {useEffect} from 'react';
import _filter from 'lodash/filter';
import {injectIntl} from 'react-intl';
import PropTypes from 'prop-types';

const propTypes = {
    /* passed props */
    setLikeData: PropTypes.func.isRequired,
    like: PropTypes.object,
    likeData: PropTypes.object,
    profileName: PropTypes.string,
    actionName: PropTypes.string,
    /* from injectIntl */
    intl: PropTypes.object.isRequired,
};

const UsersDislikeLike = (props) => {
    const {like, typeLike, profileName, likeData, intl, actionName, setLikeData} = props;
    const count = likeData && (actionName === 'dislike' ? likeData.count_dislikes : likeData.count_likes);

    const deleteUser = () => {
        if (like[typeLike] === false) {
            const newLikesName = _filter(likeData.likes_names, name => name !== profileName);
            const likeDataNames = {...likeData};
            likeDataNames.likes_names = newLikesName;
            setLikeData(likeDataNames);
        }
    };

    useEffect(() => {
        deleteUser();
    }, [like[typeLike]]);

    return (likeData !== null ? <div>
        {likeData.likes_names && <div className="st-like-users-wrap">
            <div className="st-like-users-arrow-top invisible"/>
            <div className="st-like-users-names-wrap d-flex flex-column align-items-center">
                {likeData.likes_names.map((name) =>
                    <span key={name} className="st-like-users-name"
                        data-test="st-like-users-name">{name}</span>)}
                <span className="st-like-users-name">
                    {count > 10
                        ? `${intl.formatMessage({id: 'post.likeUsersMore'})} ${count - 10}`
                        : null}
                </span>
            </div>
            <div className="st-like-users-arrow-bottom"/>
        </div>}
    </div> : null);
};

UsersDislikeLike.propTypes = propTypes;

export default injectIntl(UsersDislikeLike);
