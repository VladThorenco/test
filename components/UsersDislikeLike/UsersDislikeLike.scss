@import '../../../utils/Mixins';
.st-like-users{
  position: absolute;
  .st-like-users-wrap {
    border-radius: 4px;
    position: absolute;
    top: 30px;
    bottom: 20px;
    right: 0;
    left: -70px;
    z-index: 2;
    height: auto;
    width: 170px;
    .st-like-users-name {
      @include textStyle($font-size-small, $color-white, bold);
      text-align: center;
      width: 170px;
      overflow: hidden;
      text-overflow: ellipsis;
    }
    .st-like-users-names-wrap{
      padding: $margin-large;
      width: 170px;
      background-color: black;
      opacity: 0.9;
    }
  }
  .st-like-users-arrow-top {
    display: block;
    margin: 0 auto;
    width: 0;
    height: 100%;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-bottom: 7px solid black;
    opacity: 0.9;
  }
  .st-like-users-arrow-bottom {
    top: 4px;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%) rotate(180deg);
    position: absolute;
    display: block;
    width: 0;
    height: 100%;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid black;
    opacity: 0.9;
  }
}
