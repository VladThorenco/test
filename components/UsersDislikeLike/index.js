import PropTypes from 'prop-types';
import React from 'react';
import UsersDislikeLike from './UsersDislikeLike';

const propTypes = {
    actionName: PropTypes.oneOf(['dislike', 'like']),
    likeData: PropTypes.object,
};

export const UsersDislikeLikeContainer = (props) => <UsersDislikeLike {...props} />;

UsersDislikeLikeContainer.propTypes = propTypes;

export default UsersDislikeLikeContainer;
