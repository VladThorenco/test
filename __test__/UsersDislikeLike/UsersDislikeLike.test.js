import {mountWithStore, shallowWithStore} from '../../../shallowWrapper';
import configureStore from 'redux/configureStore';
import React from 'react';
import UsersDislikeLike from '__test__/MainPage/UsersDislikeLike/UsersDislikeLike';
import { UsersDislikeLikeContainer} from '__test__/MainPage/UsersDislikeLike';

describe('UsersDislikeLike', () => {
    const props = {
        likeData: {},
        like: {},
        actionType: 'dislike',
        typeLike: 'dislike',
        actionId: 'dislike',
        profileName: 'Master High',
        actionName: 'dislike',
        setLikeData: jest.fn(),
    };
    const initialState = { entities: {}, ui: {} };
    const store = configureStore(initialState);
    it('renders without crashing', () => {
        const component = shallowWithStore(<UsersDislikeLike {...props} />, store);
        expect(component).toMatchSnapshot();
    });
    it('testing UsersDislikeLikeContainer', () => {
        const component = mountWithStore(<UsersDislikeLikeContainer {...props} />, store);
        const findComponent = component.find('UsersDislikeLike');
        expect(findComponent).toHaveLength(1);
    });
    it('testing deleteUser calls in UseEffect', () => {
        props.like = {
            dislike: false,
        };
        mountWithStore(<UsersDislikeLike {...props} />, store);
        expect(props.setLikeData).toHaveBeenCalledTimes(1);
    });
    it('Testing are names in popup', () => {
        props.likeData = {
            likes_names: ['andrey', 'vlad'],
        };
        const component = mountWithStore(<UsersDislikeLike {...props} />, store);
        const findComponent = component.find('UsersDislikeLike');
        const items = findComponent.find('[data-test="st-like-users-name"]');
        expect(items).toHaveLength(2);
    });
});
