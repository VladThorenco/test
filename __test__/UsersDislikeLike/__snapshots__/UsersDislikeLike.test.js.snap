// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersDislikeLike renders without crashing 1`] = `ShallowWrapper {}`;
